src/providers/azure.rs

```diff
@@ -261,13 +267,13 @@ impl super::Launcher for RegionLauncher {
         self.log = Some(l.log);
         let log = self.log.as_ref().unwrap();
         let max_wait = l.max_wait;
-        let vms: Result<Vec<(String, String, _)>, Error> = l.machines
+        let vms: Result<Vec<(String, IpInfo, _)>, Error> = l.machines
             .into_iter()
             .map(|(nickname, desc)| {
                 let vm_name = super::rand_name_sep("vm", "-");
                 debug!(log, "setting up azure instance"; "nickname" => &nickname, "vm_name" => &vm_name);
 
-                let pub_ip = azcmd::create_vm(
+                let ipinfo = azcmd::create_vm(
```

Comment:
nit: `ip_info` per https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#general-conventions-[rfc-#430]

> In CamelCase, acronyms count as one word: use Uuid rather than UUID. In snake_case, acronyms are lower-cased: is_xid_start.